src/Stack/Build/Execute.hs

```diff
@@ -661,6 +660,38 @@ executePlan' installedMap0 targets plan ee@ExecuteEnv {..} = do
                   $ Map.elems
                   $ planUnregisterLocal plan
 
+unregisterSinglePackage ::
+       (HasProcessContext env, HasLogFunc env)
+    => ActualCompiler
+    -> Path Abs Dir
+    -> GhcPkgId
+    -> PackageIdentifier
+    -> Text
+    -> RIO env ()
+unregisterSinglePackage cv localDB gid ident reason = do
+    let wc = view whichCompilerL cv
+    logInfo $
+        fromString (packageIdentifierString ident) <> ": unregistering" <>
+        if T.null reason
+            then ""
+            else " (" <> RIO.display reason <> ")"
+    unregisterGhcPkgId wc cv localDB gid ident
```

Comment:
With the change to the `case` above, this whole function could simply call `unregisterPackages` like

```haskell
  unregisterPackages wc localDB [(gid, (ident, reason))]
```